lua/bible-verse/core/init.lua

```diff
@@ -110,4 +111,32 @@ function M.query_and_insert()
 	Ui:input(input_conf, on_submit)
 end
 
+--- Returns a random verse
+function M.random_query()
+	return process_query(random_verse(), Config.options.query_format, 0)
+end
+
+--- Query and shows a random verse
+function M.query_and_show_random()
+	local popup_conf = vim.deepcopy(Config.options.ui.popup)
+	local relative_size = Utils.get_relative_size(popup_conf.relative)
+
+	local popup_width = math.ceil(
+		math.min(
+			relative_size.width * popup_conf.size.max_width_percentage,
+			relative_size.width * popup_conf.size.width_percentage
+		)
+	)
+	local popup_max_height = math.ceil(relative_size.height * popup_conf.size.max_height_percentage)
+
+	local query_result = M.random_query()
```

Comment:
Need to pass in the `popup_width` such that the outputted format is correctly shown.